docs/how-to/gpu-enabled-mpi.rst

```diff
@@ -0,0 +1,260 @@
+interfaces for various network programming models and interfaces. UCX uses ROCm technologies to
+implement various network operation primitives. UCX is the standard communication library for
+InfiniBand and RDMA over Converged Ethernet (RoCE) network interconnect. To optimize data
+transfer operations, many MPI libraries, including Open MPI, can leverage UCX internally.
+
+To set up UCX so that it can be used to compile Open MPI, refer to the following instructions.
+
+1. Set environment variables to install all software components in the same base directory. We use the
+home directory in our example, but you can specify a different location if you want.
+
+    .. code-block:: shell
+
+        export INSTALL_DIR=$HOME/ompi_for_gpu
+        export BUILD_DIR=/tmp/ompi_for_gpu_build
+        mkdir -p $BUILD_DIR
+
+2. Instal UCX. To view UCX and ROCm version compatibility, refer to the
+`communication libraries tables <https://rocm.docs.amd.com/en/latest/reference/gpu_libraries/communication.html>`_.
+
+    .. code-block:: shell
+
+        export UCX_DIR=$INSTALL_DIR/ucx
+        cd $BUILD_DIR
+        git clone https://github.com/openucx/ucx.git -b v1.15.x
+        cd ucx
+        ./autogen.sh
+        mkdir build
+        cd build
+        ../configure -prefix=$UCX_DIR \
+            --with-rocm=/opt/rocm
+        make -j $(nproc)
+        make -j $(nproc) install
+
+3. Install Open MPI.
+
+    .. code-block:: shell
+
+        export OMPI_DIR=$INSTALL_DIR/ompi
+        cd $BUILD_DIR
+        git clone --recursive https://github.com/open-mpi/ompi.git \
+            -b v5.0.x
+        cd ompi
+        ./autogen.pl
+        mkdir build
+        cd build
+        ../configure --prefix=$OMPI_DIR --with-ucx=$UCX_DIR \
+            --with-rocm=/opt/rocm
+        make -j $(nproc)
+        make install
+
+.. _rocm-enabled-osu:
+
+ROCm-enabled OSU
+---------------------------------------------------------------------------------------------------------------
+
+You can use OSU Micro Benchmarks (OMB) to evaluate the performance of various primitives on
+ROCm-supported AMD GPUs. The ``--enable-rocm`` option exposes this functionality.
+
+.. code-block:: shell
+
+    export OSU_DIR=$INSTALL_DIR/osu
+    cd $BUILD_DIR
+    wget http://mvapich.cse.ohio-state.edu/download/mvapich/osu-micro-benchmarks-7.2.tar.gz
+    tar xfz osu-micro-benchmarks-7.2.tar.gz
+    cd osu-micro-benchmarks-7.2
+    ./configure --enable-rocm \
+        --with-rocm=/opt/rocm \
+        CC=$OMPI_DIR/bin/mpicc CXX=$OMPI_DIR/bin/mpicxx \
+        LDFLAGS="-L$OMPI_DIR/lib/ -lmpi -L/opt/rocm/lib/ \
+        $(hipconfig -C) -lamdhip64" CXXFLAGS="-std=c++11"
+    make -j $(nproc)
+
+Intra-node run
+----------------------------------------------------------------------------------------------------------------
+
+Before running an Open MPI job, you must set the following environment variables to ensure that
+you're using the correct versions of Open MPI and UCX.
```

Comment:
you are

Reply:
I'm intentionally using contractions to make content less formal/more approachable.

Reply:
Sounds good. Please add it as a guideline.